src/overlays/actors/ovl_En_Stream/z_en_stream.c

```diff
@@ -30,29 +29,122 @@ const ActorInit En_Stream_InitVars = {
     (ActorFunc)EnStream_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80965B20[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_VEC3F_DIV1000(scale, 20, ICHAIN_STOP),
 };
 
-#endif
-
-extern InitChainEntry D_80965B20[];
-
-extern UNK_TYPE D_06000950;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stream/EnStream_SetupAction.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stream/EnStream_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stream/EnStream_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stream/func_809656D4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stream/func_809657F4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stream/func_8096597C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stream/EnStream_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Stream/EnStream_Draw.s")
+extern Gfx D_06000950[];
+
+void EnStream_SetupAction(EnStream* this, EnStreamActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void EnStream_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnStream* this = THIS;
+
+    this->size = EN_STREAM_SIZE(&this->actor);
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    if (this->size != EN_STREAM_SIZE_NORMAL && this->size == EN_STREAM_SIZE_SMALL) {
+        this->actor.scale.y = 0.01f;
+    }
```

Comment:
This if-condition seems strange. Shouldn't `this->size == EN_STREAM_SIZE_SMALL` be enough since by being small, it's by default not `EN_STREAM_SIZE_NORMAL`?

Reply:
Yeah, it's definitely weird. But you need both to match, both in OoT and MM. I'm not sure what exactly that suggests, though. Even in OoT, this if-statement is never true in the final game.